static/app/views/starfish/modules/databaseModule/queries.ts

```diff
@@ -514,6 +519,61 @@ export const useQueryTransactionByTPMAndP75 = (
   });
 };
 
+export const useQueryGetProfileIds = (
+  transactionNames: string[],
+  spanHash: string
+): DefinedUseQueryResult<{transaction_id: string}[]> => {
+  const location = useLocation();
+  const {slug: orgSlug} = useOrganization();
+  const eventView = EventView.fromNewQueryWithLocation(
+    {
+      fields: ['transaction'],
+      name: 'Db module - profile',
+      query: `transaction:[${transactionNames.join(',')}] has:profile.id`,
+      projects: [1],
+      version: 1,
+    },
+    location
+  );
+  const discoverResult = useDiscoverQuery({eventView, location, orgSlug});
+
+  const transactionIds = discoverResult?.data?.data?.map(d => d.id);
+
+  const query = `
+    SELECT
+      transaction_id
+    FROM
+      default.spans_experimental_starfish
+    WHERE
+      group_id = '${spanHash}' AND
+      transaction_id IN ('${transactionIds?.join(`','`)}')
+  `;
+
+  return useQuery({
+    enabled: !!transactionIds?.length,
+    queryKey: ['transactionsWithProfiles', transactionIds?.join(',')],
+    queryFn: () => fetch(`${HOST}/?query=${query}`).then(res => res.json()),
+    retry: false,
+    initialData: [],
+  });
+};
+
+export const useQueryGetEvent = (
+  transactionEventId?: string
+): DefinedUseQueryResult<EventTransaction> => {
+  const path = `/api/0/projects/sentry/sentry/events/${transactionEventId?.replaceAll(
+    '-',
+    ''
+  )}/`;
+  return useQuery({
+    enabled: !!transactionEventId,
+    queryKey: ['event', transactionEventId],
+    queryFn: () => fetch(path).then(res => res.json()),
+    retry: false,
+    initialData: {},
+  });
+};
```

Comment:
I'll have to update https://github.com/getsentry/sentry/pull/48782/files#diff-24cc6c7856b7527becdc0dc545f4c1951b33a08af859e02f679aebde5c8997dcR351-R362 to use this, didn't realize there was an `enabled` parameter

Reply:
👍 Makes sense!
Enabled is good with dependent queries, so the query won't execute until enabled is true.